### PR TITLE
Update the readme

### LINKED ISSUE #219: Updates to README
https://github.com/orgsync/active_interaction/blob/v2.0.0/README.md

- [x] Make `default` option for `Array` and `Hash` more explicit. Both of them either take `nil` (making the input optional) or `[]` or `{}`, respectively (making the input default to the defaults defined in its block).
- [x] Move the validation section out of [the section about defining an interaction](https://github.com/orgsync/active_interaction/blob/v2.0.0/README.md#how-do-i-define-an-interaction). I think it should live in its own section. It could be adjacent to a new section about callbacks.
- [x] Create a section about callbacks.
- [x] We should mention that the date/time inputs don't work with the `format` option if time zones are available.
- [x] The [hash filter](https://github.com/orgsync/active_interaction/blob/v2.0.0/README.md#additional-filter-options-3) incorrectly states that all keys are symbolized. That's not since #215.
- [x] Rename `json` to something else, maybe `data` (in the interface filter).
- [x] Create a section about symbolic errors.
- [ ] Create a section about using `desc`.
- [x] Remove changelog, contributing, and license sections.
- [ ] Explain `#to_model`.
- [x] The predicate methods generated for inputs.
- [ ] Explain that `new` can take args.
- [x] Add the complete controller example from https://github.com/orgsync/active_interaction/issues/220#issuecomment-56574824.

### PR DESCRIPTION
**Pull request by tfausak**
Fixes #219.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,264 +1,1061 @@
...
+    else
+      errors.merge!(thing.errors)
+    end
+  end
+end
+```
+
+### Forms
+
+TODO
```
Comment (tfausak) [inline README.md]: @AaronLasseigne can you fill out this section?

**Comment by AaronLasseigne** [comment]
As is, I'm not a fan of the new logo. I like the idea we just need a better quality version.

**Comment by tfausak** [comment]
I agree. We need @tylerlee to make us a better version. 



It's pretty obvious a developer made that.

**Comment by tfausak** [comment]
Should we go ahead and merge this without @tylerlee's logo?

**Pull request merged**